### PR TITLE
Add typing for  _CudaStreamBase and _CudaEventBase classes

### PR DESCRIPTION
**Pull request by malfet**

### CODE REVIEW

File: torch/_C/__init__.pyi.in [hunk 1]
```diff
@@ -226,3 +229,29 @@ class _CudaDeviceProperties:
...
+    def priority_range(self) -> Tuple[_int, _int]: ...
+
+# Defined in torch/csrs/cuda/Event.cpp
+class _CudaEventBase:
+    device: _device
+    cuda_event: _int
+
+    def __new__(cls, enable_timing: _bool = False, blocking: _bool = False, interprocess: _bool = False) -> _CudaEventBase: ...
+    @classmethod
+    def from_ipc_handle(cls, device: _device, ipc_handle: Union[bytes, str]) -> _CudaEventBase: ...
```
Comment (rgommers) [inline torch/_C/__init__.pyi.in:251]: Looks like this should be only `bytes`, and I can't find a usage of it with `str` in the code base. Why do you need `str` here?

File: torch/_C/__init__.pyi.in [hunk 2]
```diff
@@ -226,3 +229,29 @@ class _CudaDeviceProperties:
     total_memory: _int
     is_integrated: _int
     is_multi_gpu_board: _int
+
+# Defined in torch/csrs/cuda/Stream.cpp
```
Comment (rgommers) [inline torch/_C/__init__.pyi.in:233]: Very minor: typo, should be `csrc`, both here and in comment further down.

**Review by rgommers — COMMENTED** [review]
LGTM, only two minor comments. Thanks @malfet

File: torch/_C/__init__.pyi.in (hunk 2 above)
Comment (malfet) [reply torch/_C/__init__.pyi.in:233]: @rgommers, thank you very much for the thorough review. Updated.

File: torch/_C/__init__.pyi.in (hunk 1 above)
Comment (malfet) [reply torch/_C/__init__.pyi.in]: I agree with you, that this call is only made with `bytes` handle, but underlying C++ implementation accepts both:
https://github.com/pytorch/pytorch/blob/b0324a97f5f5ae597114f4814482cc6bfdc17eda/torch/csrc/cuda/Event.cpp#L52
As one can easily test by invoking:
```
torch._C._CudaEventBase.from_ipc_handle(torch.device('cuda'), 'abba'*16)
```

**Review by seemethere — APPROVED** [review]

**Pull request closed without merging**